### DISCUSSION TITLE
Allow pasting into password field when signing in

### DISCUSSION THREAD

**Discussion by bustinbung** [ppy/osu#27356] [Ideas] [answered]
I use a password manager to generate long, random passwords for my accounts. osu! is no exception, my account password is around 25 random characters. When I sign in to a new osu! client, I can't copy and paste my password into the password field on signin. I'm not sure how this could be resolved or if it's platform specific (running Arch Linux now), but it would be a nice QOL improvement at least for me.

**Comment by peppy** [comment]
works on windows / macos *shrug* linux

**Reply by bustinbung** [reply]
Upon further investigation, it appears that it is working correctly on both MacOS and Linux. The issue is that the keyboard shortcut for paste is not bound to the correct key. For context, I use the [Dvorak layout](https://en.wikipedia.org/wiki/Dvorak_keyboard_layout#/media/File:KB_United_States_Dvorak.svg), so for me, <kbd>C-v</kbd> looks like <kbd>C-.</kbd> on QWERTY layouts. If I press what looks like <kbd>C-v</kbd> on QWERTY layouts, it does work (although this is <kbd>C-k</kbd> for me). I've created a table that hopefully illustrates my point.

Dvorak | QWERTY
-----|------
<kbd>C-v</kbd> | <kbd>C-.</kbd>
<kbd>C-k</kbd> | <kbd>C-v</kbd>

However, typing in other text boxes does not use the incorrect layout, I can still type in Dvorak. Because the issue is the same on both MacOS and Linux, I get the feeling that system or OS-specific keyboard input/driver issues are not the cause. Not exactly sure what the cause might be.

**Reply by peppy** [reply, marked as answer]
https://github.com/ppy/osu-framework/pull/5790

**Reply by bustinbung** [reply]
Thanks for the quick response, I appreciate it!
